### PR TITLE
updated state machine for new behaviour

### LINKED ISSUE #12: Review unlimited Job Configs
Currently Mallus will review a Job Config and, if accepted, will assume that it will be working on the job so will not listen to another Job Config until it has done the job. 

This is an issue as the DCL will flood its active connected nodes with Job Configs to establish which nodes can be added to the cluster. 

Mallus should change by:

-  Reviewing Job Configs until a Job is started
or
- Adding a new messsage telling the model it is in a cluster
- This will mean Mallus will review Job Configs until it gets the confirmation message

### PR DESCRIPTION
**Pull request by FreddieBrown**
Updated the state machine so that it works with the current sybl behaviour

### CODE REVIEW

File: sybl/client/sybl_client.py [hunk 1]
```diff
@@ -220,8 +226,12 @@ def _heartbeat(self) -> None:
...
             self._state = State.READ_JOB
             self._message_stack.append(response)
+        elif "Dataset" in response.keys():
+            logger.info("RECIEVED DATASET")
+            self._state = State.PROCESSING
+            self._message_stack.append(response)
 
-    def _process_job_config(self) -> bool:
+    def _process_job_config(self) -> None:
         while self._state == State.READ_JOB:
```
Comment (G-Kemp101) [inline sybl/client/sybl_client.py]: This no longer needs to be a loop as it will only complete a single iteration

**Review by G-Kemp101 — COMMENTED** [review]
In `_process_job` from your previous PR - i didn't notice it then - it `prints` the training and predict data. Can we avoid using `print` statements and change it to logging. Also we don't really want to be printing out whole datasets so a more useful print message is probably advised

**Comment by FreddieBrown** [comment]
> In `_process_job` from your previous PR - i didn't notice it then - it `prints` the training and predict data. Can we avoid using `print` statements and change it to logging. Also we don't really want to be printing out whole datasets so a more useful print message is probably advised

I'd argue that having them as debug statements is still useful. Could only print the first 5 rows maybe?

File: sybl/client/sybl_client.py [hunk 2]
```diff
@@ -168,50 +168,49 @@ def load_model(self, email: str, model_name: str) -> None:
...
+        predict_rids = None
 
-            if "record_id" in train_pd.columns:
-                # Take record ids from training set
-                train_pd = train_pd.drop(["record_id"], axis=1)
-                print("Training Data: {}".format(train_pd))
+        if "record_id" in train_pd.columns:
+            # Take record ids from training set
+            train_pd = train_pd.drop(["record_id"], axis=1)
+            logger.debug("Connected")("Training Data: {}".format(train_pd))
```
Comment (G-Kemp101) [inline sybl/client/sybl_client.py]: This might be a typo

**Review by G-Kemp101 — COMMENTED** [review]
First 5 rows will be fine, really don't want a whole dataset being printed

File: sybl/client/sybl_client.py (hunk 2 above)
Comment (G-Kemp101) [reply sybl/client/sybl_client.py]: Think its been picked up by MyPy

Comment (FreddieBrown) [reply sybl/client/sybl_client.py]: ah shit that is what it is, yeah that is my bad

**Review by G-Kemp101 — APPROVED** [review]

**Pull request merged**